### PR TITLE
Nginx timeouts

### PR DESCRIPTION
**Pull request by matherit**
*Issue #, if available:*

*Description of changes:*
Adds support for configuring `proxy_read_timeout` in Nginx via an environment variable. This also respects the recently added `SAGEMAKER_GUNICORN_TIMEOUT_SECONDS` to ensure upstream timeouts are avoided.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docker/build_artifacts/sagemaker/serve.py [hunk 1]
```diff
@@ -67,6 +67,16 @@ def __init__(self):
         self._gunicorn_timeout_seconds = int(
             os.environ.get("SAGEMAKER_GUNICORN_TIMEOUT_SECONDS", 30)
         )
+        self._nginx_proxy_read_timeout_seconds = int(
+            os.environ.get("SAGEMAKER_NGINX_PROXY_READ_TIMEOUT_SECONDS", 60))
+
+        # Nginx proxy read timeout should not be less than the GUnicorn timeout. If it is, this
+        # can result in upstream timed out errors.
+        if self._gunicorn_timeout_seconds > self._nginx_proxy_read_timeout_seconds:
+            self._nginx_proxy_read_timeout_seconds = self._gunicorn_timeout_seconds
```
Comment (satishpasumarthi) [inline docker/build_artifacts/sagemaker/serve.py:76]: nit: Can we also log the `nginx_proxy_read_timeout` values before and after as that is getting modified here?

**Review by satishpasumarthi — COMMENTED** [review]
Left few comments.

**Review by satishpasumarthi — APPROVED** [review]
LGTM !

File: docker/build_artifacts/sagemaker/serve.py [hunk 2]
```diff
@@ -67,6 +67,23 @@ def __init__(self):
...
+                "GUnicorn timeout was higher than Nginx proxy read timeout."
+                " Setting Nginx proxy read timeout from {} seconds to {} seconds"
+                " to match GUnicorn timeout.".format(
+                    self._nginx_proxy_read_timeout_seconds, self._gunicorn_timeout_seconds
+                )
+            )
+            self._nginx_proxy_read_timeout_seconds = self._gunicorn_timeout_seconds
+            log.info("GUnicorn timeout was higher than Nginx proxy read timeout."
+                     " Setting Nginx proxy read timeout to match GUnicorn: "
+                     "{} seconds.".format(self._nginx_proxy_read_timeout_seconds))
```
Comment (mseth10) [inline docker/build_artifacts/sagemaker/serve.py:86]: nit: Are we logging this information twice?

**Review by mseth10 — APPROVED** [review]
LGTM

File: docker/build_artifacts/sagemaker/serve.py (hunk 2 above)
Comment (matherit) [reply docker/build_artifacts/sagemaker/serve.py]: Oops. Looks like I forgot to push that removal... Fixed now.

**Pull request merged**